extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/vpc_flow_log_unmarshaler.go

```diff
@@ -0,0 +1,396 @@
+	}
+
+	return logs
+}
+
+// addToLogs parses the log line and creates
+// a new record log. The record log is added
+// to the scope logs of the resource identified
+// by the resourceKey created from the values.
+func (v *vpcFlowLogUnmarshaler) addToLogs(
+	scopeLogsByResource map[resourceKey]plog.ScopeLogs,
+	fields []string,
+	logLine string,
+) error {
+	// first line includes the fields
+	// TODO Replace with an iterator starting from go 1.24:
+	// https://pkg.go.dev/strings#FieldsSeq
+	values := strings.Split(logLine, " ")
+	nFields := len(fields)
+	nValues := len(values)
+	if nFields != nValues {
+		return fmt.Errorf("expect %d fields per log line, got log line with %d fields", nFields, nValues)
+	}
+
+	// create new key for resource and new
+	// log record to add to the scope of logs
+	// of the resource
+	key := &resourceKey{}
+	record := plog.NewLogRecord()
+
+	// There are 4 fields for the addresses: srcaddr, pkt-srcaddr,
+	// dstaddr, pkt-dstaddr. We will save these fields in a
+	// map, so we can use the right conventions in the end.
+	//
+	// See https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-records-examples.html#flow-log-example-nat.
+	ips := make(map[string]string, 4)
+
+	// range over the fields of the log line
+	for i, field := range fields {
+		value := values[i]
+		if value == "-" {
+			// If a field is not applicable or could not be computed for a
+			// specific record, the record displays a '-' symbol for that entry.
+			//
+			// See https://docs.aws.amazon.com/vpc/latest/userguide/flow-log-records.html.
+			continue
+		}
+
+		if strings.HasPrefix(field, "ecs-") {
+			v.logger.Warn("currently there is no support for ECS fields")
+			continue
+		}
+
+		found, err := handleField(field, value, record, ips, key)
+		if err != nil {
+			return err
+		}
+		if !found {
+			v.logger.Warn("field is not an available field for a flow log record",
+				zap.String("field", field),
+				zap.String("documentation", "https://docs.aws.amazon.com/vpc/latest/userguide/flow-log-records.html"),
+			)
+		}
+	}
+
+	// get the address fields
+	addresses := handleAddresses(ips)
+	for field, value := range addresses {
+		record.Attributes().PutStr(field, value)
+	}
+
+	scopeLogs := v.getScopeLogs(*key, scopeLogsByResource)
+	rScope := scopeLogs.LogRecords().AppendEmpty()
+	record.MoveTo(rScope)
+
+	return nil
+}
+
+// handleAddresses creates a new map where the original field
+// names will be the known conventions for the fields
+func handleAddresses(addresses map[string]string) map[string]string {
+	// max is 3 fields, see example in
+	// https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-records-examples.html#flow-log-example-nat
+	recordAttr := make(map[string]string, 3)
+	srcaddr, foundSrc := addresses["srcaddr"]
+	pktSrcaddr, foundSrcPkt := addresses["pkt-srcaddr"]
+	if !foundSrcPkt && foundSrc {
+		// there is no middle layer, and "srcaddr" field
+		// corresponds to the original source address.
```

Comment:
```suggestion
		// there is no middle layer, assume "srcaddr" field
		// corresponds to the original source address.
```

It might not be the original source address, but this is the best we can do with the information we have